dist: drop scylla-jmx

syuu1228 opened the pull request:
Since Java based tools and JMX server are deprecated, drop them from submodule, build system and package definition.

Related scylladb/scylla-tools-java#370
Related #14856

syuu1228 commented:
I need clarify:
 - Can we drop java and jmx submodules now?
 - I move nodetools-wrapper from scylla-tools to scylla-core, should we add similar wrapper for sstable commands? Or, maybe we move to `scylla <command>` style and won't need the wrapper?

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7647/)
#### Build Details:
- Duration: 32 min
- Builder: spider2.cloudius-systems.com

denesb commented:
> I need clarify:
> 
>     * Can we drop java and jmx submodules now?


Not just yet, we need https://github.com/scylladb/scylla-ccm/pull/565 to go in first. Dtests still use the java tools (via ccm).

>     * I move nodetools-wrapper from scylla-tools to scylla-core, should we add similar wrapper for sstable commands? Or, maybe we move to `scylla <command>` style and won't need the wrapper?

We should just remove the `nodetool-wrapper`. We cannot even add similar wrappers for the sstable tools, because its interface and output is not compatible with the Java sstable tools.

syuu1228 commented:
Dropped nodetool-wrapper.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7685/)
#### Build Details:
- Duration: 42 min
- Builder: spider1.cloudius-systems.com

denesb commented:
@syuu1228 we will need something like this too:
```diff
diff --git a/bin/nodetool b/bin/nodetool
new file mode 100755
index 0000000000..968457f0c6
--- /dev/null
+++ b/bin/nodetool
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+SCRIPT_PATH=$(dirname $(realpath "$0"))
+
+INSTALLED_SCYLLA_PATH="${SCRIPT_PATH}/scylla"
+
+# Allow plugging scylla path for local testing
+exec ${SCYLLA:-${INSTALLED_SCYLLA_PATH}} nodetool $@
diff --git a/install.sh b/install.sh
index de6aaee167..bc6ad8a440 100755
--- a/install.sh
+++ b/install.sh
@@ -422,6 +422,7 @@ for bin in libexec/*; do
     adjust_bin "${bin#libexec/}"
 done
 install -m644 ubsan-suppressions.supp -Dt "$rprefix/libexec"
+install -m755 bin/nodetool "$rprefix/bin"
 
 install -d -m755 "$rdoc"/scylla
 install -m644 README.md -Dt "$rdoc"/scylla/
diff --git a/scripts/create-relocatable-package.py b/scripts/create-relocatable-package.py
index c7b6f65fb4..e30f8bd43f 100755
--- a/scripts/create-relocatable-package.py
+++ b/scripts/create-relocatable-package.py
@@ -171,6 +171,7 @@ ar.reloc_add('swagger-ui')
 ar.reloc_add('api')
 ar.reloc_add('tools/scyllatop')
 ar.reloc_add('scylla-gdb.py')
+ar.reloc_add('bin/nodetool')
 ar.reloc_add(args.debian_dir, arcname='debian')
 node_exporter_dir = args.node_exporter_dir
 if args.stripped:
 ```
 
 This is just a scratch I used for testing, feel free to change/refine.

denesb commented:
Hi @syuu1228 I think we should start with a smaller-scope change, dropping just JMX first.
I ended up reducing the scope of https://github.com/scylladb/scylla-ccm/pull/565 to prepare for JMX being dropped.
Once this goes in, we can start preparing to drop the java tools package:
* We will need to add a `bin/nodetool` to `scylla.git`, something like I showed above, at the same time, we have to drop `scylla-tools-java.git/bin/nodetool-wrapper`.
* We need to figure out how ccm/dtest will get cassandra-stress once we drop the java tools. Maybe we leave the `tools/java` submodule for now but reduce it to install just `cassandra-stress` (optionally).

I will also start cleaning dtests of any remaining java tools usage.

syuu1228 commented:
> Hi @syuu1228 I think we should start with a smaller-scope change, dropping just JMX first. I ended up reducing the scope of [scylladb/scylla-ccm#565](https://github.com/scylladb/scylla-ccm/pull/565) to prepare for JMX being dropped.

Rebased patch to just drop JMX.

avikivity commented:
We need separate release pipelines first.

yaronkaikov commented:
> We need separate release pipelines first.

Created  https://github.com/scylladb/scylla-pkg/issues/4037 for tools-java and https://github.com/scylladb/scylla-pkg/issues/4038 for JMX.

yaronkaikov commented:
> > We need separate release pipelines first.
> 
> Created [scylladb/scylla-pkg#4037](https://github.com/scylladb/scylla-pkg/issues/4037) for `tools-java` and [scylladb/scylla-pkg#4038](https://github.com/scylladb/scylla-pkg/issues/4038) for `JMX`

we will start with the JMX part i guess, so we can get this PR in

syuu1228 commented:
I need to modify Obsoletes/Breaks/Replaces on this patch, since I supposed next release version is 5.5 so I wrote something like `Obsoletes:      %{product}-jmx < 5.5` in this patch, but now we already released 6.0.

avikivity commented:
Converted to draft until we have separate scylla-jmx packaging.

tchaikov commented:
sorry, i missed this PR. will take a look tomorrow.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9234/)
#### Build Details:
- Duration: 1 hr 7 min
- Builder: i-03a5be15d075b22a6 (m5d.12xlarge)

syuu1228 commented:
Something wrong on CI:
```
20:24:56  dpkg: dependency problems prevent configuration of scylla:
20:24:56   scylla depends on scylla-tools (= 6.1.0~dev-0.20240527.f30ae0fce0f3-1); however:
20:24:56    Package scylla-tools is not installed.
20:24:56   scylla depends on scylla-tools-core (= 6.1.0~dev-0.20240527.f30ae0fce0f3-1); however:
20:24:56    Package scylla-tools-core is not installed.
20:24:56  
20:24:56  dpkg: error processing package scylla (--install):
20:24:56   dependency problems - leaving unconfigured
20:24:56  Errors were encountered while processing:
20:24:56   scylla
20:24:56  Error: while running runtime: exit status 1
```

Maybe I broke dependencies when I updated the patch, I will look into it.

tchaikov commented on dist/debian/control.template:
```diff
@@ -47,10 +47,20 @@ Conflicts: prometheus-node-exporter
 Architecture: any
 Depends: %{product}-server (= ${binary:Version})
+ , %{product}-tools (= ${binary:Version})
+ , %{product}-tools-core (= ${binary:Version})
```
why are we adding `scylla-tools` and `scylla-tools-core` back as dependencies of `scylla` ? they were previously removed in b8634fb244ab08c2ffa66700745b06e21ee57f30 .

syuu1228 replied:
This was mistakenly happened because of the patch based on older code, fixed.

tchaikov commented on dist/redhat/scylla.spec:
```diff
@@ -11,9 +11,11 @@ Requires:       %{product}-server = %{version}-%{release}
 Requires:       %{product}-python3 = %{version}-%{release}
 Requires:       %{product}-kernel-conf = %{version}-%{release}
+Requires:       %{product}-tools = %{version}-%{release}
+Requires:       %{product}-tools-core = %{version}-%{release}
```
ditto.

syuu1228 replied:
This one too, fixed.

tchaikov commented on tools/toolchain/Dockerfile:
```diff
@@ -4,6 +4,10 @@ ARG CLANG_BUILD="SKIP"
+ADD ./install-dependencies.sh ./
+ADD ./seastar/install-dependencies.sh ./seastar/
+ADD ./tools/java/install-dependencies.sh ./tools/java/
+ADD ./tools/toolchain/system-auth ./
```
probably this deserves its dedicated PR? also i cannot find the explanations regarding this change in the commit message. also, if we want to run the local script on host, probably we could use `RUN --mount=type=bind`, like:
```Dockerfile
RUN --mount=type=bind,source=.,target=./scylla dnf -y update \
    && dnf -y install 'dnf-command(copr)' \
    && dnf -y install ccache \
    && dnf -y install devscripts debhelper fakeroot file rpm-build \
    && scylla/install-dependencies.sh && dnf clean all \
    && echo 'ALL ALL=(ALL:ALL) NOPASSWD: ALL' >> /etc/sudoers \
    && cp scylla/tools/toolchain/system-auth /etc/pam.d \
    && echo 'Defaults !requiretty' >> /etc/sudoers
RUN mkdir -p /root/.m2/repository
ENV JAVA8_HOME=/usr/lib/jvm/java-1.8.0-openjdk
RUN scylla/tools/toolchain/optimized_clang.sh
```

unless we need to access these scripts in the container after building the image.

syuu1228 replied:
Also this one too, fixed.

fruch commented:
@syuu1228 

we should run offline installer test from SCT, when this gets stable.

syuu1228 commented:
Fixed mentioned code, and rebased with latest master.
Need to check the CI can pass, since previously it failed.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10090/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10090/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10385/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2360/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10090/)
#### Build Details:
- Duration: 2 hr 26 min
- Builder: spider2.cloudius-systems.com

syuu1228 commented:
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10090/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10090/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10385/) ❌ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2360/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10090/)
> 
> #### Build Details:
> * Duration: 2 hr 26 min
> * Builder: spider2.cloudius-systems.com

Seems like it's getting Java related error on dtest:
```
04:02:57  ---------------------------- live log sessionstart -----------------------------
04:02:57  19:02:56,318 root            ERROR common.py      :1020 | Command '['bash', '-c', '/home/jenkins/.ccm/scylla-repository/17969/scylla-core-package/install.sh --prefix /home/jenkins/.ccm/scylla-repository/17969 --nonroot --supervisor --without-systemd']' returned non-zero exit status 1.
04:02:57  19:02:56,319 root            ERROR common.py      :1020 | stdout:
04:02:57  b'Please install openjdk-8 or openjdk-11 before running install.sh.\n'
04:02:57  19:02:56,319 root            ERROR common.py      :1020 | stderr:
04:02:57  b'/home/jenkins/.ccm/scylla-repository/17969/scylla-core-package/install.sh: line 110: scylla-jmx/select-java: No such file or directory\n'
04:02:57  INTERNALERROR> Traceback (most recent call last):
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/site-packages/_pytest/main.py", line 268, in wrap_session
04:02:57  INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/site-packages/pluggy/_hooks.py", line 513, in __call__
04:02:57  INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
04:02:57  INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
04:02:57  INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
04:02:57  INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/site-packages/pluggy/_callers.py", line 182, in _multicall
04:02:57  INTERNALERROR>     return outcome.get_result()
04:02:57  INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/site-packages/pluggy/_result.py", line 100, in get_result
04:02:57  INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/site-packages/pluggy/_callers.py", line 103, in _multicall
04:02:57  INTERNALERROR>     res = hook_impl.function(*args)
04:02:57  INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/jenkins/workspace/scylla-master/gating-dtest-release/scylla-dtest/conftest.py", line 365, in pytest_sessionstart
04:02:57  INTERNALERROR>     _scylla_mode = scylla_mode(cassandra_dir, scylla_version)
04:02:57  INTERNALERROR>                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/jenkins/workspace/scylla-master/gating-dtest-release/scylla-dtest/tools/marks.py", line 35, in scylla_mode
04:02:57  INTERNALERROR>     return dtest_config.get_scylla_mode()
04:02:57  INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/jenkins/workspace/scylla-master/gating-dtest-release/scylla-dtest/dtest_config.py", line 90, in get_scylla_mode
04:02:57  INTERNALERROR>     ccm_repo_cache_dir, _ = ccmlib.scylla_repository.setup(self.scylla_version)
04:02:57  INTERNALERROR>                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
04:02:57  INTERNALERROR>   File "/home/jenkins/.local/lib/python3.12/site-packages/ccmlib/scylla_repository.py", line 381, in setup
04:02:57  INTERNALERROR>     run_scylla_unified_install_script(**args)
04:02:57  INTERNALERROR>   File "/home/jenkins/.local/lib/python3.12/site-packages/ccmlib/scylla_repository.py", line 609, in run_scylla_unified_install_script
04:02:57  INTERNALERROR>     run('''{0}/install.sh --prefix {1} --nonroot{2}'''.format(
04:02:57  INTERNALERROR>   File "/home/jenkins/.local/lib/python3.12/site-packages/ccmlib/scylla_repository.py", line 42, in run
04:02:57  INTERNALERROR>     subprocess.run(['bash', '-c', cmd], cwd=cwd, check=True, capture_output=True)
04:02:57  INTERNALERROR>   File "/usr/local/lib/python3.12/subprocess.py", line 571, in run
04:02:57  INTERNALERROR>     raise CalledProcessError(retcode, process.args,
04:02:57  INTERNALERROR> subprocess.CalledProcessError: Command '['bash', '-c', '/home/jenkins/.ccm/scylla-repository/17969/scylla-core-package/install.sh --prefix /home/jenkins/.ccm/scylla-repository/17969 --nonroot --supervisor --without-systemd']' returned non-zero exit status 1.
04:02:57  Docker exitcode: 3
```
https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10385/console

avikivity commented:
Well it's failing install.sh. Need to drop jmx from there too.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10226/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10226/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2471/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10496/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10226/)
#### Failed Tests (3/31138):
- [test_content[redis]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10226/testReport/junit/system_tables_test/TestProtocolServersTable/Tests___dtest_with_topology_changes___test_content_redis_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_content[redis])
- [test_content[redis]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10226/testReport/junit/system_tables_test/TestProtocolServersTable/Tests___dtest___test_content_redis_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_content[redis])
- [test_removenode_rejoin](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10226/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___dtest_with_topology_changes___test_removenode_rejoin) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_removenode_rejoin)
#### Build Details:
- Duration: 2 hr 27 min
- Builder: spider4.cloudius-systems.com

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10260/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10260/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2501/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10526/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10260/)
#### Failed Tests (4/31112):
- [test_content[redis]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10260/testReport/junit/system_tables_test/TestProtocolServersTable/Tests___dtest_with_topology_changes___test_content_redis_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_content[redis])
- [test_removenode_rejoin](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10260/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___dtest_with_topology_changes___test_removenode_rejoin) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_removenode_rejoin)
- [test_content[redis]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10260/testReport/junit/system_tables_test/TestProtocolServersTable/Tests___dtest___test_content_redis_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_content[redis])
- [test_gossip_boot.dev.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10260/testReport/junit/%28root%29/test_gossip_boot/Tests___Unit_Tests___test_gossip_boot_dev_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_gossip_boot.dev.2)
#### Build Details:
- Duration: 4 hr 44 min
- Builder: i-0f6435adf2834d14a (m5ad.12xlarge)

syuu1228 commented:
Rebased with latest master, and also dropped "Obsoletes: scylla-jmx" from both rpm/deb, because I thought we completely drop scylla-jmx package on next release, but we actually decided to drop scylla-jmx from default installation but keep it as optional.
A side effect of this is that upgrading from an existing version will leave you with the old scylla-jmx package.
We may need to document to uninstall it manually.
(Previous code was intended to uninstall scylla-jmx manually, but then user cannot manually install scylla-jmx anymore since it conflicts with scylla metapackage)

syuu1228 commented:
@yaronkaikov I think we can stop Draft mode now, right?

yaronkaikov commented:
@denesb @tchaikov Can you review ?

avikivity commented:
Do we have an external jmx package? Let's add document it in docs/ so people can find it.

syuu1228 commented:
> Do we have an external jmx package? Let's add document it in docs/ so people can find it.

No not yet, but once we add new tag on scylla-jmx reposotory, Github Actions will publish "release" page on GitHub page.
We probably just need to update version number on SCYLLA-VERSION-GEN in scylla-jmx repository, then create new tag using the new version number.
Once we publish "release" page we can add the document and link to scylla-jmx GitHub URL.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11338/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11338/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/346/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3395/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11338/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11418/)
#### Failed Tests (1/34991):
- [test_concurrent_restart](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11338/testReport/junit/nodetool_additional_test/TestNodetool/Tests___dtest___test_concurrent_restart) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_concurrent_restart)
#### Build Details:
- Duration: 3 hr 37 min
- Builder: spider5.cloudius-systems.com

tchaikov reviewed (approved):
lgtm. but Avi insist that we need to have separate release pipelines *first* and have separate a scylla-jmx packaging. so we might want to address his concerns first.

fruch commented:
> lgtm. but Avi insists that we need to have separate release pipelines *first* and have separate a scylla-jmx packaging. so we might want to address his concerns first.

Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.

denesb commented:
> > lgtm. but Avi insists that we need to have separate release pipelines _first_ and have separate a scylla-jmx packaging. so we might want to address his concerns first.
> 
> Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.

I think it is to keep supporting the odd user who still uses the JMX interface directly to manage Scylla.

yaronkaikov commented:
> > > lgtm. but Avi insists that we need to have separate release pipelines _first_ and have separate a scylla-jmx packaging. so we might want to address his concerns first.
> > 
> > 
> > Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.
> 
> I think it is to keep supporting the odd user who still uses the JMX interface directly to manage Scylla.

We already have a separate release flow, @syuu1228 implemented using GitHub actions. I think this one can be merged

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11346/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11346/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3406/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/357/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11429/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11346/)
#### Build Details:
- Duration: 6 hr 7 min
- Builder: spider7.cloudius-systems.com

fruch commented:
> > > lgtm. but Avi insists that we need to have separate release pipelines _first_ and have separate a scylla-jmx packaging. so we might want to address his concerns first.
> > 
> > 
> > Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.
> 
> I think it is to keep supporting the odd user who still uses the JMX interface directly to manage Scylla.



> > > lgtm. but Avi insists that we need to have separate release pipelines _first_ and have separate a scylla-jmx packaging. so we might want to address his concerns first.
> > 
> > 
> > Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.
> 
> I think it is to keep supporting the odd user who still uses the JMX interface directly to manage Scylla.

seems like quite an effort for this unknown user.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3412/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11435/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/363/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/)
#### Failed Tests (4/35034):
- [test_resharding_by_smp_decrease[4-TimeWindowCompactionStrategy-15]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/resharding_test/TestReshardingVariants/Tests___dtest_with_topology_changes___test_resharding_by_smp_decrease_4_TimeWindowCompactionStrategy_15_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_resharding_by_smp_decrease[4-TimeWindowCompactionStrategy-15])
- [test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-LeveledCompactionStrategy]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/compaction_additional_test/TestCompactionAdditional/Tests___dtest_with_topology_changes___test_refresh_and_restart_after_compaction_strategy_change_SizeTieredCompactionStrategy_LeveledCompactionStrategy_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-LeveledCompactionStrategy])
- [test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-TimeWindowCompactionStrategy]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/compaction_additional_test/TestCompactionAdditional/Tests___dtest_with_topology_changes___test_refresh_and_restart_after_compaction_strategy_change_SizeTieredCompactionStrategy_TimeWindowCompactionStrategy_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-TimeWindowCompactionStrategy])
- [test_replace_with_background_workload[use_endpoint-rbo_disabled]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/replace_address_test/TestReplaceAddress/Tests___dtest_with_topology_changes___test_replace_with_background_workload_use_endpoint_rbo_disabled_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_replace_with_background_workload[use_endpoint-rbo_disabled])
#### Build Details:
- Duration: 3 hr 1 min
- Builder: spider7.cloudius-systems.com

yaronkaikov commented:
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/) ❌ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3412/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11435/) ✅ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/363/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/)
> 
> #### Failed Tests (4/35034):
> * [test_resharding_by_smp_decrease[4-TimeWindowCompactionStrategy-15]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/resharding_test/TestReshardingVariants/Tests___dtest_with_topology_changes___test_resharding_by_smp_decrease_4_TimeWindowCompactionStrategy_15_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_resharding_by_smp_decrease%5B4-TimeWindowCompactionStrategy-15%5D)
> * [test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-LeveledCompactionStrategy]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/compaction_additional_test/TestCompactionAdditional/Tests___dtest_with_topology_changes___test_refresh_and_restart_after_compaction_strategy_change_SizeTieredCompactionStrategy_LeveledCompactionStrategy_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_refresh_and_restart_after_compaction_strategy_change%5BSizeTieredCompactionStrategy-LeveledCompactionStrategy%5D)
> * [test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-TimeWindowCompactionStrategy]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/compaction_additional_test/TestCompactionAdditional/Tests___dtest_with_topology_changes___test_refresh_and_restart_after_compaction_strategy_change_SizeTieredCompactionStrategy_TimeWindowCompactionStrategy_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_refresh_and_restart_after_compaction_strategy_change%5BSizeTieredCompactionStrategy-TimeWindowCompactionStrategy%5D)
> * [test_replace_with_background_workload[use_endpoint-rbo_disabled]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/replace_address_test/TestReplaceAddress/Tests___dtest_with_topology_changes___test_replace_with_background_workload_use_endpoint_rbo_disabled_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_replace_with_background_workload%5Buse_endpoint-rbo_disabled%5D)
> 
> #### Build Details:
> * Duration: 3 hr 1 min
> * Builder: spider7.cloudius-systems.com

@syuu1228 I guess this failure is not related to this change, right ?

syuu1228 commented:
> > #### 🔴 CI State: FAILURE
> > ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/) ❌ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3412/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11435/) ✅ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/363/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11361/)
> > #### Failed Tests (4/35034):
> > 
> > * [test_resharding_by_smp_decrease[4-TimeWindowCompactionStrategy-15]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/resharding_test/TestReshardingVariants/Tests___dtest_with_topology_changes___test_resharding_by_smp_decrease_4_TimeWindowCompactionStrategy_15_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_resharding_by_smp_decrease%5B4-TimeWindowCompactionStrategy-15%5D)
> > * [test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-LeveledCompactionStrategy]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/compaction_additional_test/TestCompactionAdditional/Tests___dtest_with_topology_changes___test_refresh_and_restart_after_compaction_strategy_change_SizeTieredCompactionStrategy_LeveledCompactionStrategy_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_refresh_and_restart_after_compaction_strategy_change%5BSizeTieredCompactionStrategy-LeveledCompactionStrategy%5D)
> > * [test_refresh_and_restart_after_compaction_strategy_change[SizeTieredCompactionStrategy-TimeWindowCompactionStrategy]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/compaction_additional_test/TestCompactionAdditional/Tests___dtest_with_topology_changes___test_refresh_and_restart_after_compaction_strategy_change_SizeTieredCompactionStrategy_TimeWindowCompactionStrategy_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_refresh_and_restart_after_compaction_strategy_change%5BSizeTieredCompactionStrategy-TimeWindowCompactionStrategy%5D)
> > * [test_replace_with_background_workload[use_endpoint-rbo_disabled]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11361/testReport/junit/replace_address_test/TestReplaceAddress/Tests___dtest_with_topology_changes___test_replace_with_background_workload_use_endpoint_rbo_disabled_) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_replace_with_background_workload%5Buse_endpoint-rbo_disabled%5D)
> > 
> > #### Build Details:
> > 
> > * Duration: 3 hr 1 min
> > * Builder: spider7.cloudius-systems.com
> 
> @syuu1228 I guess this failure is not related to this change, right ?

No, I don't think this is related with this change

avikivity commented:
> > > > lgtm. but Avi insists that we need to have separate release pipelines _first_ and have separate a scylla-jmx packaging. so we might want to address his concerns first.
> > > 
> > > 
> > > Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.
> > 
> > 
> > I think it is to keep supporting the odd user who still uses the JMX interface directly to manage Scylla.
> 
> We already have a separate release flow, @syuu1228 implemented using GitHub actions. I think this one can be merged

We need documentation pointing to the jmx repository somewhere, with a reference in docs/getting-started/install-scylla/index.rst.

syuu1228 commented:
> > > > > lgtm. but Avi insists that we need to have separate release pipelines _first_ and have separate a scylla-jmx packaging. so we might want to address his concerns first.
> > > > 
> > > > 
> > > > Can anyone point to the issue/document discussing this ? I can understand why we need to keep jmx alive, for what purpose.
> > > 
> > > 
> > > I think it is to keep supporting the odd user who still uses the JMX interface directly to manage Scylla.
> > 
> > 
> > We already have a separate release flow, @syuu1228 implemented using GitHub actions. I think this one can be merged
> 
> We need documentation pointing to the jmx repository somewhere, with a reference in docs/getting-started/install-scylla/index.rst.

Okay I will update the document, but we don't have any releases on scylla-jmx Github page, yet.
To get the release, please also merge update version number PR on https://github.com/scylladb/scylla-jmx/pull/243, and also please create new tag for the version: scylla-6.2.0.
When new tag pushed, Github Actions will automatically publish the release page.

syuu1228 commented:
I just found scylla_setup is referencing scylla-jmx package, dropped it:
https://github.com/scylladb/scylladb/compare/dfc41df4bafbe3755ce8dbd620d4b3af14a5c4cd..5eb5b444fae965c3ae7ffe94aa20f4d400ecc6db

scylladb-promoter commented:
## Docs Preview :book:
Docs Preview for this pull request is available [here](http://previews.docs.scylladb.com/scylladb/17969)

**Changed Files:**
- [getting-started/install-scylla/index.html](http://previews.docs.scylladb.com/scylladb/17969/getting-started/install-scylla/index.html)
- [getting-started/install-scylla/install-jmx](http://previews.docs.scylladb.com/scylladb/17969/getting-started/install-scylla/install-jmx)
- [getting-started/install-scylla/install-on-linux](http://previews.docs.scylladb.com/scylladb/17969/getting-started/install-scylla/install-on-linux)
- [getting-started/installation-common/unified-installer](http://previews.docs.scylladb.com/scylladb/17969/getting-started/installation-common/unified-installer)
- [upgrade/upgrade-opensource/upgrade-guide-from-6.1-to-6.2/upgrade-guide-from-6.1-to-6.2-generic](http://previews.docs.scylladb.com/scylladb/17969/upgrade/upgrade-opensource/upgrade-guide-from-6.1-to-6.2/upgrade-guide-from-6.1-to-6.2-generic)
- [upgrade/upgrade-opensource/upgrade-guide-from-6.x.y-to-6.x.z](http://previews.docs.scylladb.com/scylladb/17969/upgrade/upgrade-opensource/upgrade-guide-from-6.x.y-to-6.x.z)

**Note:** This preview will be available for 30 days and will be automatically deleted after that period. You can manually trigger a new build by committing changes.

syuu1228 commented:
> We need documentation pointing to the jmx repository somewhere, with a reference in docs/getting-started/install-scylla/index.rst.

Ah, I just pushed doument update but I mistakenly added the document on this repository, not jmx repositry.
I will move it to jmx repositry.

avikivity commented:
> > We need documentation pointing to the jmx repository somewhere, with a reference in docs/getting-started/install-scylla/index.rst.
> 
> Ah, I just pushed doument update but I mistakenly added the document on this repository, not jmx repositry. I will move it to jmx repositry.

What's wrong with the update on this repository? We need to leave a pointer from here to the new home of scylla-jmx, since users will begin by looking at the standard scylladb documentation.

syuu1228 commented:
> > What's wrong with the update on this repository? We need to leave a pointer from here to the new home of scylla-jmx, since users will begin by looking at the standard scylladb documentation.
> 
> I just reviewed it, it looks fine to me.

Okay, then I keep current version.
So can we merge this PR now?

avikivity commented:
> > > What's wrong with the update on this repository? We need to leave a pointer from here to the new home of scylla-jmx, since users will begin by looking at the standard scylladb documentation.
> > 
> > 
> > I just reviewed it, it looks fine to me.
> 
> Okay, then I keep current version. So can we merge this PR now?

Not as long as CI fails.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11461/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11461/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/448/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3496/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11519/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11461/)
#### Failed Tests (2/34976):
- [test_parallel_operations_for_10ks_and_10tables_and_clearallsnapshots](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11461/testReport/junit/snapshot_test/TestParallelSnapshotOperations/Tests___dtest_with_tablets___test_parallel_operations_for_10ks_and_10tables_and_clearallsnapshots) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_parallel_operations_for_10ks_and_10tables_and_clearallsnapshots)
- [test_removenode_rejoin](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11461/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___dtest_with_topology_changes___test_removenode_rejoin) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_removenode_rejoin)
#### Build Details:
- Duration: 6 hr 27 min
- Builder: i-04e367ac8c13d8cb9 (m5ad.8xlarge)

syuu1228 commented:
Rebased with latest master

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11560/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11560/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/508/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11579/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3562/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11560/)
#### Build Details:
- Duration: 3 hr 13 min
- Builder: spider5.cloudius-systems.com

yaronkaikov commented:
@denesb can we merge this?

denesb commented:
> > * [test_error_message_for_batch_exceeding_fail_threshold.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11527/testReport/junit/%28root%29/test_logs/Tests___Unit_Tests___test_error_message_for_batch_exceeding_fail_threshold_debug_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_error_message_for_batch_exceeding_fail_threshold.debug.1)
> > * [test_error_message_for_batch_exceeding_fail_threshold.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11527/testReport/junit/%28root%29/test_logs/Tests___Unit_Tests___test_error_message_for_batch_exceeding_fail_threshold_debug_1_2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_error_message_for_batch_exceeding_fail_threshold.debug.1)
> > * [cql-pytest.test_logs.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11527/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___cql_pytest_test_logs_debug_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+cql-pytest.test_logs.debug.1)
> 
> > #### 🔴 CI State: FAILURE
> > ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11527/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11527/) ✅ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/490/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/11561/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/3543/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11527/)
> > #### Failed Tests (3/35261):
> > ```
> > * [test_error_message_for_batch_exceeding_fail_threshold.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11527/testReport/junit/%28root%29/test_logs/Tests___Unit_Tests___test_error_message_for_batch_exceeding_fail_threshold_debug_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_error_message_for_batch_exceeding_fail_threshold.debug.1)
> > 
> > * [test_error_message_for_batch_exceeding_fail_threshold.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11527/testReport/junit/%28root%29/test_logs/Tests___Unit_Tests___test_error_message_for_batch_exceeding_fail_threshold_debug_1_2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_error_message_for_batch_exceeding_fail_threshold.debug.1)
> > 
> > * [cql-pytest.test_logs.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/11527/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___cql_pytest_test_logs_debug_1) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+cql-pytest.test_logs.debug.1)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > #### Build Details:
> > ```
> > * Duration: 3 hr 25 min
> > 
> > * Builder: spider1.cloudius-systems.com
> > ```
> 
> There is a coredump here, it is probably unrelated. I will to have a look and create an issue for it, in the meanwhile I re-kicked CI.

Clarification (after having a look):
* There are many coredumps (22), but all are from intentional crashes produced by error injection.
* There is a SEGFAULT causing the crash which in turn fails the tests. There is no coredump for this sadly, nor any information to go on.

The SEGFAULT:
```
AddressSanitizer:DEADLYSIGNAL                                                                                                                                                                                                                                  
=================================================================                                                                                                                                                                                              
==114852==ERROR: AddressSanitizer: SEGV on unknown address 0x000100000000 (pc 0x7f89de610a8f bp 0x7f88cb3fd290 sp 0x7f88cb3fd230 T0)                                                                                                                           
==114852==The signal is caused by a READ memory access.                                                                                                                                                                                                        
AddressSanitizer:DEADLYSIGNAL                                                                                                                                                                                                                                  
AddressSanitizer: nested bug in the same thread, aborting.                                                                                                                                                                                                     
------ Ending test cql-pytest.test_logs.1 ------  
```

lersek commented:
@tchaikov @syuu1228 I think this PR (commit 3cd2a6173668) may have caused a regression for the scylla-ci build in jenkins *if* the "use cmake" parameter is enabled. Cursorily looking at the commit, I think the jmx dependency was not removed from the cmake infrastructure.

https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11783/consoleFull

~~~
[2024-09-20T15:41:10.196Z] [3000/4164] Generating /jenkins/workspace/scylla-master/scylla-ci/scylla/tools/jmx/build/scylla-jmx-6.3.0~dev-0.20240920.d011dd30216f.noarch.tar.gz
[2024-09-20T15:41:10.197Z] FAILED: /jenkins/workspace/scylla-master/scylla-ci/scylla/tools/jmx/build/scylla-jmx-6.3.0~dev-0.20240920.d011dd30216f.noarch.tar.gz
[2024-09-20T15:41:10.197Z] cd /jenkins/workspace/scylla-master/scylla-ci/scylla/tools/jmx && reloc/build_reloc.sh --version scylla-6.3.0~dev-0.20240920.d011dd30216f --nodeps
[2024-09-20T15:41:10.197Z] /bin/sh: line 1: reloc/build_reloc.sh: No such file or directory
~~~

I can reproduce the issue locally (when using `./configure --use-cmake`), with `ninja -C build dist:Dev`.

The following patch seems to fix it:
~~~
diff --git a/dist/CMakeLists.txt b/dist/CMakeLists.txt
index 1c66b6f759b7..948df8a8e75b 100644
--- a/dist/CMakeLists.txt
+++ b/dist/CMakeLists.txt
@@ -110,8 +110,6 @@ set(dist_pkgs
 dist_submodule(cqlsh cqlsh dist_pkgs)
 dist_submodule(tools java dist_pkgs
   NOARCH)
-dist_submodule(jmx jmx dist_pkgs
-  NOARCH)
 dist_submodule(python3 python3 dist_pkgs)
 set(unified_dist_pkg
   "${CMAKE_BINARY_DIR}/$<CONFIG>/dist/tar/${Scylla_PRODUCT}-unified-${Scylla_VERSION}-${Scylla_RELEASE}.${CMAKE_SYSTEM_PROCESSOR}.tar.gz")
@@ -134,7 +132,6 @@ add_custom_target(dist)
 add_dependencies(dist
   dist-cqlsh
   dist-tools
-  dist-jmx
   dist-python3
   dist-server)

diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index c4c3b5675a91..a307322a849f 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -26,8 +26,6 @@ include(build_submodule)
 build_submodule(cqlsh cqlsh)
 build_submodule(tools java
   NOARCH)
-build_submodule(jmx jmx
-  NOARCH)

 execute_process(
   COMMAND "${CMAKE_SOURCE_DIR}/install-dependencies.sh" --print-python3-runtime-packages
diff --git a/tools/testing/dist-check/CMakeLists.txt b/tools/testing/dist-check/CMakeLists.txt
index 4d02e797cc06..7d045698830c 100644
--- a/tools/testing/dist-check/CMakeLists.txt
+++ b/tools/testing/dist-check/CMakeLists.txt
@@ -13,6 +13,5 @@ add_custom_target(dist-check
 add_dependencies(dist-check
   dist-server-rpm
   dist-python3-rpm
-  dist-jmx-rpm
   dist-tools-rpm
   dist-cqlsh-rpm)
~~~

tchaikov commented:
@lersek thank you for identifying the issue and providing the patch, i created #20736 with your patch. and added your Signed-off-by tag, hopefully it's fine by you.
